README.md

```diff
@@ -5,38 +5,88 @@
-- [React DOM](https://facebook.github.io/react/) (tested with 15.3.1)
-- [Underscore](http://underscorejs.org/)  (tested with 1.8.3)
+ReactDom.render(
+    <Badge label='Hello react-vapor!' />, 
+    document.getElementById('SomewhereInYourApp')
+);
+```
+In a `.tsx` context: 
+```tsx
+import * as React from 'react';
+import * as ReactDom from 'react-dom';
+import {Badge} from 'react-vapor';
 
-### Building
-Make sure you have Node JS and NPM installed.
-Run `npm install` to get the required dependencies and build the librairy.
+ReactDom.render(
+    <Badge label='Hello react-vapor!' />, 
+    document.getElementById('SomewhereInYourApp')
+);
+```
+----
+## Contributing
 
-### Running documentation locally
-To build and run the doc locally, simply run `npm start` and access http://localhost:8080/.
+### Build
 
-### Running the tests suite
-Run `npm test` to run all tests and get the code coverage!
+Clone the repository using your favorite method.
+```bash
+git clone https://github.com/coveo/react-vapor.git
+# or
+git clone git@github.com:coveo/react-vapor.git
+```
 
-## Contributing
-1. Search the issues, if it is not already there, add one.
-2. Fork the repository
-3. Code Code Code
-4. Submit a pull request
-5. Wait for some nice guy to review and merge
+Make sure you have Node JS and NPM installed and install `react-vapor`'s dependencies.
+```bash
+npm install
+# or 
+yarn install
+```
+
+### Run the demo page locally
+
+To build and run the demo page locally, simply run `npm start` and access http://localhost:8080/. Note that the demo page will refresh automatically on file save.
+
+### Make sure everything works
+Running the complete test suite will ensure your local `react-vapor` build is healthy.
+```bash
+npm test
+```
+
+### Guidelines
+* We aim to have the highest code coverage possible. A pull request that increases or unchanges the code coverage ratio has higher chances of getting merged quickly.
+* Make sure your pull requests follow our code style. Normally the `lintfix` npm script is run on pre-commit so this should not be a problem. However, you can double check by running the script manually with `npm run lintfix`.
+* We're pretty flexible on the commit messages structure and pull requests descriptions, but still make sure you give enough details to your fellow reviewers.
+* We tend to avoid comments in our code base, we strongly prefer good naming and code structure.
+
+### Tips and tricks
+* `npm test` command creates a local coverage report at the end of its execution that you can browse through in order to see the hits and misses of your unit tests. View the report by opening up the `coverage/lcov-report/index.html` file in your favorite internet browser.
+* `npm run test:watch` command will run tests on file save.
+* `npm run reconstruct` command will reinstall and rebuild from scratch your local branch without loosing any pending change. Run this if you think your current state is messed up.

```

Comment:
will reinstall and rebuild from scratch your local branch -> will reinstall and rebuild your local branch from scratch
loosing -> discarding
is messed up -> is corrupted